deepdiver1975/tarstreamer/src/TarStreamer.php

```diff
@@ -0,0 +1,308 @@
+<?php
+
+namespace ownCloud\TarStreamer;
+
+class TarStreamer {
+
+	const REGTYPE = 0;
+	const DIRTYPE = 5;
+	const XHDTYPE = 'x';
+
+	/**
+	 * Process in 1 MB chunks
+	 */
+	protected $blockSize = 1048576;
+	protected $outStream;
+	protected $needHeaders = false;
+
+	/**
+	 * Create a new TarStreamer object.
+	 *
+	 * @param array $options
+	 */
+	public function __construct($options = []){
+		if (isset($options['outstream'])){
+			$this->outStream = $options['outstream'];
+		} else {
+			$this->outStream = fopen('php://output', 'w');
+			// turn off output buffering
+			while (ob_get_level() > 0){
+				ob_end_flush();
+			}
+		}
+	}
+
+	/**
+	 * Send appropriate http headers before streaming the tar file and disable output buffering.
+	 * This method, if used, has to be called before adding anything to the tar file.
+	 *
+	 * @param string $archiveName Filename of archive to be created (optional, default 'archive.tar')
+	 * @param string $contentType Content mime type to be set (optional, default 'application/x-tar')
+	 */
+	public function sendHeaders($archiveName = 'archive.tar', $contentType = 'application/x-tar'){
+		$encodedArchiveName = rawurlencode($archiveName);
+		if (headers_sent($headerFile, $headerLine)){
+			die(
+					"<p><strong>Error:</strong> Unable to send file " .
+					"$encodedArchiveName. HTML Headers have already been sent from " .
+					"<strong>$headerFile</strong> in line <strong>$headerLine" .
```

Comment:
I don't feel to happy about this as it may leak unnecessary information.

Can we not just throw an exception? - This way we could also log the error in ownCloud properly.